includes/class-wcpdf-admin.php

```diff
@@ -1300,47 +1300,57 @@ public function handle_heartbeat_received( array $response, array $data ): array
 		return $response;
 	}
 
+	/**
+     * Updates documents data in the "PDF document data" meta box if the generation in the background is finished.
+     *
+	 * @return void
+	 */
 	public function ajax_fetch_pdf_document_data(): void {
-		if ( isset( $_REQUEST['nonce'] ) && wp_verify_nonce( $_REQUEST['nonce'], 'generate_wpo_wcpdf' ) ) {
+		if ( ! isset( $_REQUEST['nonce'] ) || wp_verify_nonce( $_REQUEST['nonce'], 'generate_wpo_wcpdf' ) ) {
 			wp_send_json_error( array(
-				'message' => esc_html__( 'Nonce expired!', 'woocommerce-pdf-invoices-packing-slips' ),
+				'message' => esc_html__( 'Invalid or expired nonce!', 'woocommerce-pdf-invoices-packing-slips' ),
 			) );
 		}
 
-		if ( empty( $_REQUEST['document_types'] ) || empty( $_REQUEST['order_id'] ) ) {
+		if ( empty( $_REQUEST['document_types'] ) || empty( (int) $_REQUEST['order_id'] ) ) {
 			wp_send_json_error( array(
-				'message' => esc_html__( 'Incomplete request!', 'woocommerce-pdf-invoices-packing-slips' ),
+				'message' => esc_html__( 'Incomplete or incorrect request!', 'woocommerce-pdf-invoices-packing-slips' ),
 			) );
 		}
 
-		$document_types  = $_REQUEST['document_types'];
+		$order_id        = (int) $_REQUEST['order_id'];
+		$document_types  = array_map( 'sanitize_text_field', $_REQUEST['document_types'] );
 		$data['invoice'] = array(
 			'number'           => array( 'label' => __( 'Invoice number:', 'woocommerce-pdf-invoices-packing-slips' ) ),
 			'date'             => array( 'label' => __( 'Invoice date:', 'woocommerce-pdf-invoices-packing-slips' ) ),
 			'display_date'     => array( 'label' => __( 'Invoice display date:', 'woocommerce-pdf-invoices-packing-slips' ) ),
 			'creation_trigger' => array( 'label' => __( 'Invoice created via:', 'woocommerce-pdf-invoices-packing-slips' ) ),
 			'notes'            => array( 'label' => __( 'Notes (printed in the invoice):', 'woocommerce-pdf-invoices-packing-slips' ) ),
 		);
-		$data            = apply_filters( 'wpo_wcpdf_ajax_fetch_pdf_document_data_fields', $data, $_REQUEST['order_id'] );
+		$data            = apply_filters( 'wpo_wcpdf_ajax_fetch_pdf_document_data_fields', $data, $order_id );
 
+		ob_start();
 		foreach ( $document_types as $document_type ) {
-			if ( ! in_array( $document_type, array_keys( $data ), true ) ) {
+			if ( ! wpo_wcpdf_is_document_type_valid( $document_type ) || ! in_array( $document_type, array_keys( $data ), true ) ) {
 				continue;
 			}
 
-			$document = wcpdf_get_document( $document_type, wc_get_order( $_REQUEST['order_id'] ) );
+			$document = wcpdf_get_document( $document_type, wc_get_order( $order_id ) );
 			if ( $document && $document->exists() ) {
-				ob_start();
 				$this->output_number_date_edit_fields( $document, $data[ $document_type ] );
-				$documents_data[ $document_type ] = ob_get_clean();
+				$documents_data[ $document_type ] = ob_get_contents();
```

Comment:
I think you missed my comment here: https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/pull/757#discussion_r1548165448